consensus/istanbul/validator/default_test.go

```diff
@@ -206,3 +209,76 @@ func testStickyProposer(t *testing.T) {
 		t.Errorf("proposer mismatch: have %v, want %v", val, val2)
 	}
 }
+
+func testQuorumSize(t *testing.T) {
```

Comment:
I think you can simplify this test by:
(1) removing the checks on the valSet size (because that's not really what this unit test is about)
(2) in a while loop for a reasonable upper limit on valSet.Size() (say, N <= 1000), check with returned value of the QuorumSize() function against an invariant that is known to be correct. For example, the formula ceil(2N/3) is derived from the invariant QuorumSize > valSet.Size - QuorumSize + f. So I recommend using that instead to make the test both more robust and succinct.

Reply:
done. yes that makes sense.